<a href="https://albertomerani.org/">
    <img src="https://drive.google.com/uc?export=view&id=16uZ9pqgmRVKrXRhDO_acSUe0ZJAfuy1C" width="1000" align="center">
</a>


<h1>Introducción a Python</h1>

<p><strong>Bienvenido!</strong> Este cuaderno te enseñará los fundamentos del lenguaje de programación Python. Aunque la información presentada aquí es bastante básica, constituye una base importante que te ayudará a leer y escribir código en Python. Al final de este cuaderno, conocerás los conceptos básicos de Python, incluyendo cómo escribir comandos básicos, comprender algunos tipos fundamentales y realizar operaciones simples en ellos.</p>


<h2>Tabla de contenidos</h2>
<div class="alerta alerta-bloque alerta-info" estilo="margin-top: 20px">
     <ul>
         <li>
             <a href="#hello">Di "Hola mundo" en Python</a>
             <ul>
                 <li><a href="#version">¿Qué versión de Python estamos usando?</a></li>
                 <li><a href="#comments">Escribir comentarios en Python</a></li>
                 <li><a href="#errors">Errores en Python</a></li>
                 <li><a href="#python_error">¿Python sabe de tu error antes de ejecutar tu código?</a></li>
                 <li><a href="#ejercise">Ejercicio: su primer programa</a></li>
             </ul>
         </li>
         <li>
             <a href="#types_objects">Tipos de objetos en Python</a>
             <ul>
                 <li><a href="#int">Enteros</a></li>
                 <li><a href="#float">Float</a></li>
                 <li><a href="#convert">Conversión de un tipo de objeto a otro tipo de objeto diferente</a></li>
                 <li><a href="#bool">Tipo de datos booleanos</a></li>
                 <li><a href="#exer_type">Ejercicio: Tipos</a></li>
             </ul>
         </li>
         <li>
             <a href="#expressions">Expresiones y variables</a>
             <ul>
                 <li><a href="#exp">Expresiones</a></li>
                 <li><a href="#exer_exp">Ejercicio: Expresiones</a></li>
                 <li><a href="#var">Variables</a></li>
                 <li><a href="#exer_exp_var">Ejercicio: Expresión y Variables en Python</a></li>
             </ul>
         </li>
     </ul>
</div>

<hr>

<h2 id="hello">Decir 'Hola mundo!' en Python</h2>


Cuando nos aventuramos en el aprendizaje de un nuevo idioma de programación, es una práctica común comenzar con un ejemplo que podríamos llamar "hello world". Por más elemental que parezca, esta única línea de código actúa como el timbre que nos confirma que sabemos cómo mostrar un mensaje en la salida y cómo ejecutar comandos en celdas de un cuaderno.

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
[Consejo]: Para ejecutar el código Python en la siguiente celda de código, haga clic en la celda para seleccionarla y presione <kbd>Shift</kbd> + <kbd>Enter</kbd>.</div>
<hr/>

In [6]:
print("¡Hola mundo!")

¡Hola mundo!


<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
    [Consejo:] <code>print()</code> es una función. Pasaste la cadena <code>'¡Hola mundo!'</code> como argumento para indicarle a Python qué imprimir.
</div>
<hr/>

<h3 id="version">¿Qué versión de Python estamos usando?</h3>

In [5]:
import sys
print(sys.version)

3.14.1 (tags/v3.14.1:57e0d17, Dec  2 2025, 14:05:07) [MSC v.1944 64 bit (AMD64)]


<h3 id="comments">Escribir comentarios en Python</h3>

<p>
    Los comentarios son líneas de texto que el intérprete de Python ignora. Se utilizan para explicar el código o para dejar notas para otros programadores. Los comentarios se pueden escribir en una línea o en varias líneas.

Para escribir un comentario en una línea, use el símbolo #. Por ejemplo, el siguiente código imprime la cadena "¡Hola mundo!" y luego agrega un comentario que explica lo que hace el código:</p>


In [4]:

print("¡Hola mundo!") # Imprime "¡Hola mundo!" en la pantalla


¡Hola mundo!


<p>
    Para escribir un comentario en varias líneas, use tres comillas simples (''') o tres comillas dobles ("""). Por ejemplo, el siguiente código imprime la cadena "¡Hola mundo!" y luego agrega un comentario que explica lo que hace el código en varias líneas:
</p>

In [7]:
"""
Este código imprime la cadena "¡Hola mundo!" en la pantalla.

Los comentarios se pueden escribir en una línea o en varias líneas.
Para escribir un comentario en una línea, use el símbolo `#`.
Para escribir un comentario en varias líneas, use tres comillas simples (`'''`) o 3 comillas dobles.
"""

print("¡Hola mundo!")

¡Hola mundo!


<h3 id="errors">Errores en Python
</h3>

<p>Los errores son problemas que ocurren cuando el intérprete de Python no puede entender o ejecutar el código. Hay dos tipos de errores: errores de sintaxis y errores de ejecución.

Los errores de sintaxis ocurren cuando el código no está escrito correctamente. Por ejemplo, el siguiente código contiene un <code>SyntaxError</code> porque el paréntesis de cierre falta:</p>

In [9]:
# Print string as error message

print("¡Hola mundo!")

¡Hola mundo!


<p>El mensaje de error le dice:
<ol>
     <li>dónde ocurrió el error (más útil en secuencias de comandos o celdas grandes de cuadernos) y</li>
     <li>qué tipo de error fue (NameError)</li>
</ol>

</p>

<p>Los errores de ejecución ocurren cuando el código se ejecuta correctamente, pero hay un problema con el código en sí. Por ejemplo, el siguiente código contiene un error de ejecución porque intenta dividir por cero: </p>

In [11]:
x = 1 / 1

Python es lo que se llama un <em>lenguaje interpretado</em>. Los lenguajes compilados examinan todo su programa en el momento de la compilación y pueden advertirle sobre toda una clase de errores antes de la ejecución. Por el contrario, Python interpreta su script línea por línea mientras lo ejecuta. Python dejará de ejecutar todo el programa cuando encuentre un error (a menos que el programador espere y maneje el error, un tema más avanzado que cubriremos más adelante en este curso).

In [12]:
saludo = "¡Hola!"
nombre = "Mundo"
print(saludo, nombre)  # Imprime múltiples cadenas de texto y una variable

fruta = "manzana"
print("Me encanta comer una deliciosa", fruta)  # Imprime una cadena con una variable

¡Hola! Mundo
Me encanta comer una deliciosa manzana


<h3 id="ejercise">Ejercicio: Tu primer programa</h3>

In [13]:
# Solicita al usuario ingresar la longitud y el ancho del rectángulo
longitud = float(input("Ingrese la longitud del rectángulo: "))
ancho = float(input("Ingrese el ancho del rectángulo: "))

# Reemplace la siguiente línea con la ecuación para calcular el área del rectángulo
area = longitud * ancho  # Reemplace 'None' con la ecuación correcta

# Imprime el resultado
print("El área del rectángulo es:", area)

El área del rectángulo es: 96.0


In [14]:
96.0

96.0

Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

# Solicita al usuario ingresar la longitud y el ancho del rectángulo
longitud = float(input("Ingrese la longitud del rectángulo: "))
ancho = float(input("Ingrese el ancho del rectángulo: "))

# Calcula el área del rectángulo
area = longitud * ancho

# Imprime el resultado
print("El área del rectángulo es:", area)

-->

<h3 id="ejercise">Ejercicio: Tu Segundo programa</h3>

<p> Imagina que eres un estudiante y deseas calcular tu promedio en tres exámenes que has realizado en una materia específica. Puedes utilizar este programa para ingresar las calificaciones de tus tres exámenes y obtener de inmediato el promedio.

Por ejemplo, ingreses las calificaciones de tus exámenes como sigue:

* Primer examen: 90
* Segundo examen: 85
* Tercer examen: 78

    El programa calculará el promedio de estas calificaciones de la siguiente manera:</p>

In [15]:
# Solicita al usuario ingresar tres números
numero1 = 90
numero2 = 85
numero3 = 78

# Calcula el promedio de los números
promedio = numero1 + numero2 + numero3 / 3 # Reemplace 'None' con la ecuación correcta

# Imprime el resultado
print("El promedio de los números ingresados es:", promedio)

El promedio de los números ingresados es: 201.0


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

# Solicita al usuario ingresar tres números
numero1 = float(input("Ingrese el primer número: "))
numero2 = float(input("Ingrese el segundo número: "))
numero3 = float(input("Ingrese el tercer número: "))

# Calcula el promedio de los números
promedio = (numero1 + numero2 + numero3) / 3

# Imprime el resultado
print("El promedio de los números ingresados es:", promedio)

-->


<hr>

<h2 id="types_objects" align="center">Tipos de objetos en Python</h2>

<p>En Python, hay varios tipos de objetos con los que puedes trabajar. Estos tipos representan diferentes tipos de datos y comprenderlos es fundamental para la programación en Python. Aquí están algunos de los tipos de objetos más comunes en Python:

    Entero (int): Representa números enteros, tanto positivos como negativos, sin punto decimal. Por ejemplo, 5, -3 y 0 son enteros.

    Flotante (float): Representa números reales con punto decimal. Por ejemplo, 3.14, -0.5 y 2.0 son flotantes.

    Cadena (str): Representa texto. Las cadenas se encierran entre comillas simples o dobles. Por ejemplo, "Hola, Mundo" y 'Python' son cadenas.

    Booleano (bool): Representa True (verdadero) o False (falso). Los booleanos se utilizan a menudo en operaciones lógicas y estructuras de control.

</p>

In [16]:
# Declarar un entero
x = 12

# Imprimir el entero
print(x)

12


In [17]:
# Declarar un float
x = 3.1416

# Imprimir el float
print(x)

3.1416


In [18]:
# Declarar un booleano
x = True

# Imprimir el booleano
print(x)

True


In [19]:
# String

# Declarar un número entero
a = '10a'

# Imprimir el número entero
print(a)

10a


<p>Puedes hacer que Python te indique el tipo de expresión utilizando la función incorporada <code>type()</code>. Notarás que Python se refiere a los números enteros como <code>int</code>, a los flotantes como <code>float</code> y a las cadenas de caracteres como <code>str</code>.</p>

In [20]:
# Type of 12

type(12)

int

In [21]:
# Type of 2.14

type(3.1416)

float

In [22]:
# Type of True

type(True)

bool

In [23]:
# Type of "Hello, Python 101!"

type('10a')

str

<p>En la celda de código siguiente, use la función <code>type()</code> para verificar el tipo de objeto de <code> '45.0'</code>.

In [24]:

print(type('45.0'))



<class 'str'>


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

type('45.0')

-->

<h3 id="convert">Convertir de un tipo de objeto a otro tipo de objeto diferente </h3>

<p>Puedes cambiar el tipo de objeto en Python; esto se llama encasillamiento. Por ejemplo, puedes convertir un <i>entero</i> en un <i>flotante</i> (por ejemplo, 2 a 2,0).</p>
<p>Probémoslo:</p>

In [25]:
# Declarar un entero
x = 452
print( type(x) )

<class 'int'>


In [26]:
# Convertir el entero a un float
x = float(x)

In [27]:
# Imprimir el float
print( type(x) )

<class 'float'>


<p>Para convertir un float a entero en Python, se puede usar la función int(). Esta función truncará los decimales del float, conservando solo la parte entera.
</p>

In [28]:
# Declarar un float
x = 1.5
print( type(x) )

# Convertir el float a un entero
y = int(x)

# Imprimir el entero
print( type(y) )

<class 'float'>
<class 'int'>


<h4>Conversión de cadenas a números enteros o flotantes</h4>

<p>A veces, podemos tener una cadena que contiene un número dentro de ella. Si este es el caso, podemos convertir esa cadena que representa un número en un número entero usando <code>int()</code>:</p>

In [29]:
# Convertir un string a un entero
x = "123"
y = int(x)
print(y)

123


<p>Pero si intentas hacerlo con una cadena que no coincide perfectamente con un número, obtendrás un error. Pruebe lo siguiente:</p>

In [31]:
# Convertir una cadena en un número entero con error

int('1')

1

<p>También puedes convertir cadenas que contengan números de punto flotante en objetos <i>flotantes</i>:</p>

In [32]:
# Convertir un string a un float
x = "123.456"
y = float(x)
print(y)

123.456


<p> Hacer el mismo proceso con punto flotante a un int causara un error</p>

In [34]:
# Convertir un string a un int
x = "123456"
y = int(x)
print(y)

123456


<h4>Convertir números en cadenas</h4>

In [35]:
# Convertir un int a un string
x = 123
y = str(x)

print(y)
print( type(y) )

123
<class 'str'>


<h3 id="bool">Boolean data type</h3>

<p><i>Booleano</i> es otro tipo importante en Python. Un objeto de tipo <i>Booleano</i> puede tomar uno de dos valores: <code>True</code> o <code>False</code>:</p>

In [36]:
# Value true

True

True

In [37]:
# Value false

False

False

<p>Cuando le pides a Python que muestre el tipo de un objeto booleano, mostrará <code>bool</code> que significa <i>booleano</i>:</p>

In [38]:
# Type of True

type(True)

bool

In [39]:
# Type of False

type(False)

bool

<p>Podemos convertir objetos booleanos a otros tipos de datos. Si convertimos un booleano con un valor de <code>True</code> a un número entero o flotante, obtendremos uno. Si convertimos un booleano con un valor de <code>False</code> a un número entero o flotante obtendremos un cero. De manera similar, si convertimos un 1 en un booleano, obtienes un <code>True</code>. Y si le ponemos un 0 a un booleano obtendremos un <code>False</code>. Probémoslo:</p>

In [40]:
# Convert True to int

int(True)

1

In [41]:
# Convert 1 to boolean

bool(1)

True

In [42]:
# Convert 0 to boolean

bool(0)

False

In [43]:
# Convert True to float

float(True)

1.0

<h3 id="exer_type">Ejercicio: Tipos</h3>

<p>¿Cuál es el tipo de datos del resultado de: <code>6 / 2</code>?</p>

In [44]:
type(6/2)

float

Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

type(6/2) # float
-->

<p>¿Cuál es el tipo de resultado de: <code>6 // 2</code>? (Tenga en cuenta la doble barra <code>//</code>.)</p>

In [45]:
type(6//2)

int

Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

type(6//2) # int, ya que las barras dobles representan división de enteros
-->

<p>Escribe un programa que imprima el tipo de cada uno de los siguientes valores:
    
* 1
* 1.0
* "Hola mundo"
* True
</p>

In [46]:
type(1)
type(1.0)
type("Hola Mundo")
type(True)

bool

Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:

# Declarar los valores
x = 1
y = 1.0
z = "Hola mundo"
w = True

# Imprimir los tipos
print(type(x))
print(type(y))
print(type(z))
print(type(w))

-->

<hr>

<h2 id="expressions">Expresión y variables</h2>

<h3 id="exp">Expresiones</h3>

<p> Una expresión es una combinación de variables, operadores y valores que produce un valor. Un valor es una expresión en sí misma.

Algunos ejemplos de expresiones son:

* 1 + 2
* "Hola mundo"
* x * y
*  z ** 2
</p>

In [47]:
# Expresión de operación de suma

# Declarar variables
x = 1
y = 2

# Calcular la suma
z = x + y

# Imprimir el resultado
print(z)

3


In [48]:
# Declarar variables
x = 10
y = 5

# Realizar la resta
z = x - y

# Imprimir el resultado
print(z)

5


<p>Podemos hacer multiplicaciones usando un asterisco:</p>

In [49]:
# Declarar variables
x = 10
y = 15

# Realizar la resta
z = x * y

# Imprimir el resultado
print(z)

150


<p>También podemos realizar la división con la barra diagonal:

In [50]:
# Division operation expression

25 / 5

5.0

In [51]:
# Division operation expression

25 / 6

4.166666666666667

<p>Como se vio en el cuestionario anterior, podemos usar la doble barra para la división de enteros, donde el resultado se redondea al entero más cercano:

In [52]:
# Integer division operation expression

25 // 5

5

In [53]:
# Integer division operation expression

25 // 6

4

* En Python, las variables se pueden declarar en cualquier lugar del código.
* Las variables se pueden usar para almacenar cualquier tipo de datos.
* Las expresiones se pueden usar para realizar cálculos y operaciones con variables.

<h3 id="exer_exp">Ejercicio: Expresión</h3>

<p>Escribamos una expresión que calcule cuántas horas hay en 160 minutos:

In [54]:
minutos = 160
horas = minutos / 60 
print(horas)

2.6666666666666665


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:
160/60
# Or
160//60
-->

<p>Python sigue convenciones matemáticas bien aceptadas al evaluar expresiones matemáticas. En el siguiente ejemplo, Python suma 30 al resultado de la multiplicación (es decir, 120).

In [55]:

30 + 2 * 60

150

<p>Y al igual que las matemáticas, las expresiones entre paréntesis tienen prioridad. Entonces lo siguiente multiplica 32 por 60.

In [56]:

(30 + 2) * 60

1920

<h3 id="var">Variables</h3>

<p>Al igual que con la mayoría de los lenguajes de programación, podemos almacenar valores en <i>variables</i>, para poder usarlos más adelante. Por ejemplo:</p>

In [57]:
# Store value into variable

x = 4 + 60 + 4 + 41 - 50

<p>Para ver el valor de <code>x</code> en un Notebook, simplemente podemos colocarlo en la última línea de una celda:</p>

In [58]:
x

59

<p>También podemos realizar operaciones en <code>x</code> y guardar el resultado en una nueva variable:</p>

In [59]:
# Usar otra variable para almacenar el resultado de la operación entre variable y valor

y = x / 60
y

0.9833333333333333

<p>Si guardamos un valor en una variable existente, el nuevo valor sobrescribirá el valor anterior:</p>

In [60]:
# Sobrescribir variable con nuevo valor
x = x / 60
x

0.9833333333333333

<h3 id="exer_exp_var">Ejercicio: Expresión y Variables en Python</h3>

<p>¿Cuál es el valor de <code>x</code> donde <code>x = 3 + 2 * 2</code></p>

In [1]:
x = 3 + 2 * 2
print(x)

7


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:
x = 3 + 2 * 2
print(x)
-->


<p>What is the value of <code>y</code> where <code>y = (3 + 2) * 2</code>?</p>

In [2]:
y = (3+2)*2
print(y)

10


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:
z = x + y
print(z)
-->

<p>What is the value of <code>z</code> where <code>z = x + y</code>?</p>

In [3]:
z = x + y
print(z)

17


Haga doble clic __aquí__ para ver la solución.

<!-- Su respuesta está a continuación:
17
-->

<hr>
<div class="alert alert-block alert-info" style="margin-top: 20px">

<h2>Felicitaciones por llegar hasta aquí!</h2>

¡Has completado una nueva hazaña!

Has aprendido sobre los tipos de datos.

Ahora eres capaz de declarar variables y hacer operaciones básicas.


<h3>¡Sigue aprendiendo y creciendo!</h3>
    
</div>

<hr>

<h2>Autores y colaborares:</h2>


<a href="https://www.linkedin.com/in/robinssondeantonio/" title="Robinsson Deantonio"> Robinsson S. Deantonio</a>

<p> Basado en aportes de:
    <a href="https://cognitiveclass.ai/" title="cognitiveclass"> cognitiveclass</a>,
    <a href="https://www.linkedin.com/in/joseph-s-50398b136/" title="cognitiveclass"> Joseph Santarcangelo</a>,
    <a href="https://www.linkedin.com/in/reevejamesd/" title="cognitiveclass"> James Reeve</a>,  
    <a href="www.linkedin.com/in/jiahui-mavis-zhou-a4537814a" title="cognitiveclass"> Mavis Zhou</a>,
</p>

<hr>
<p>Copyright &copy; 2023. Este cuaderno y su código fuente se publican según los términos del <a href="https://cognitiveclass.ai/mit-license/">MIT License</a>.</p>